modules/kok-bot/modules/karma/listener.js

```diff
@@ -10,45 +11,63 @@ client.on("message", msg => {
+                    createKarma();
+                } else {
+                    msg.channel.sendMessage("Please wait for cooldown period");
+                }
+                guildUsers.users[0].lastKarmaCreated = Date.now();
+
+                guildUsers.save()
+                    .catch(err => console.log());
+            });
+
+        function createKarma() {
+            mentioned.array().forEach((mention, index, mentions) => {
+                if (mention.id === msg.author.id) {
+                    reply.push(`Trying to give yourself karma ${mention.username}. Shame!`);
+                } else {
+                    mentionUsers.push(mention.username);
+                    let guildID = msg.guild.id;
+                    guildUsersModel.findOneOrCreate({ guildID }, { guildID })
+                        .then(guildUsers => {
+                            if (guildUsers) {
+                                let duplicate = false;
+                                guildUsers.users.forEach((user, index, users) => {
+                                    if (user.id === mention.id) {
+                                        if (!users[index].karma) {
+                                            users[index].karma = 0;
+                                        }
+                                        users[index].karma++;
+                                        duplicate = true;
                                     }
-                                    users[index].karma++;
-                                    duplicate = true;
+                                });
+                                if (!duplicate) {
+                                    guildUsers.users.push({ id: mention.id, karma: 1 });
                                 }
-                            });
-                            if (!duplicate) {
-                                guildUsers.users.push({ id: mention.id, karma: 1 });
+                                guildUsers.save()
+                                    .catch(err => {
+                                        console.log(err);
+                                    });
+                            } else {
+                                throw new Error("findOneOrCreate() has not returned a model");
                             }
-                            guildUsers.save()
-                                .catch(err => {
-                                    console.log(err);
-                                });
-                        } else {
-                            throw new Error("findOneOrCreate() has not returned a model");
-                        }
-                    })
-                    .catch(err => {
-                        console.log(err);
-                    })
-            }
-        });
+                        })
+                        .catch(err => {
+                            console.log(err);
```

Comment:
let's use the error logger for these errors so that we can easily trace them if needs be